plugins/doorbird/src/main.ts

```diff
@@ -188,43 +249,80 @@ class DoorbirdCamera extends ScryptedDeviceBase implements Intercom, Camera, Vid
         const password: string = this.getPassword();
         const audioTxUrl: string = `${this.getHttpBaseAddress()}/bha-api/audio-transmit.cgi`;
 
-        this.console.log('Doorbird: Starting audio transmitter...');
-
         (async () => {
-            this.console.log('Doorbird: audio transmitter started.');
 
+            this.console.log('Doorbird: Audio transmitter started.');
             const passthrough = new PassThrough();
-            authHttpFetch({
-                method: 'POST',
-                url: audioTxUrl,
-                credential: {
-                    username,
-                    password,
-                },
-                headers: {
-                    'Content-Type': 'audio/basic',
-                    'Content-Length': '9999999'
-                },
-                data: passthrough,
-            });
+            const abortController = new AbortController();
+            let totalBytesWritten: number = 0;
+            let totalTimeWaited: number = 0;
 
             try {
-                while (true) {
-                    const data = await readLength(socket, 1024);
+                // Perform POST request instantly instead of unneeded handling with DIGEST authentication.
+                // Credentials will be thrown into network by all other requests anyway.
+                httpFetch({
+                    url: audioTxUrl,
+                    method: 'POST',
+                    headers: {
+                        'Content-Type': 'audio/basic',
+                        'Content-Length': '9999999',
+                        'Authorization': this.getBasicAuthorization(username, password),
+                    },
+                    signal: abortController.signal,
+                    body: passthrough,
+                    responseType: 'readable',
+                })
+                // The ideal interval for 256-byte chunks at 8000 bytes/sec is 32ms.
+                const throttlingIntervalMillis: number = 32;
+
+                // Initialize the next chunk send timestamp to the current time.
+                let nextChunkSendTimeStamp: number = Date.now();
+
+                while (true) {  // Loop will be broken by StreamEndError.
+
+                    // Read the next chunk of audio data from the Doorbird camera.
+                    const data = await readLength(socket, DoorbirdCamera.TRANSMIT_AUDIO_CHUNK_SIZE);
+                    if (data.length === 0) {
+                        break;
+                    }
+
+                    // Calculate the time to wait until the next chunk should be sent.
+                    const timeToWait = Math.max(0, nextChunkSendTimeStamp - Date.now());
```

Comment:
why is sleep necessary here? the audio itself should be arriving at real time. should be able to send immediately.